### PR TITLE
[TechDocs] Adjust theme styles

### PR DESCRIPTION
**Pull request by camilaibs**
Signed-off-by: Camila Belo <camilaibs@gmail.com>

## Hey, I just made a Pull Request!

- Fixes https://github.com/backstage/backstage/issues/6289
- Style tweaks as a side effect of these changes:
   - https://github.com/backstage/mkdocs-techdocs-core/pull/50
   - https://github.com/backstage/techdocs-container/pull/29

See below how the page appearance is after changes:

| Light theme | Dark theme |
| ------------ | ----------- |
| image |  image |

## How to test?

1. Clone the [mkdocs-techdocs-core](https://github.com/backstage/mkdocs-techdocs-core/pull/50) project
2. Check out to the branch `bump-mkdocs-material`
3. Install the plugin locally:
```sh
pip install -e .
```
4. Make sure you have installed version `0.2.3`:
```sh
pip show mkdocs-techdocs-core
```
5. As we are going to test it locally, you need to have your environment configured like [here](https://github.com/backstage/techdocs-container/blob/main/Dockerfile#L21)

6. Navigate to the backstage repository folder
7. Also check out to the branch `bump-mkdocs-material`
8. Run `techdocs-cli `locally in development mode:
```sh
yarn techdocs-cli:dev serve --no-docker
```
9. Start the local embedded app in another terminal:
```sh
yarn workspace techdocs-cli-embedded-app start
```
10. Finally access the docs page:
http://localhost:3000/docs/default/component/local

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — APPROVED** [review]
🟢 to unblock, but I'm not quite sure why reviewers were ping'd 😅

**Comment by Rugvip** [comment]
Dawned on me that ofc we are ping'd, since we're listed as owners everywhere now. You should still be able to merge this without a 🟢 from reviewers though so will keep that in mind :p

File: .changeset/techdocs-moose-design.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-techdocs': minor
+---
+
+Adjust the Tech Docs page theme as a side effect of the `mkdocs-material` theme update.
```
Comment (emmaindal) [inline .changeset/techdocs-moose-design.md:5]: I wonder if we should add a requirement here that one should be on a specific version of mkdocs-material or mkdocs-techdocs-core?

**Review by ottosichert — COMMENTED** [review]
Thanks for the excellent instructions on how to test this PR!

I'm surprised to see things working with 3 major versions bumped 😄 however I noticed the resizing to smaller screens broke in a way that the sidebar overlaps with the content, see screenshot below:

Screenshot 2022-02-17 at 09 56 36

File: packages/techdocs-cli-embedded-app/src/components/TechDocsPage/TechDocsPage.tsx [hunk 2]
```diff
@@ -24,6 +39,90 @@ import {
...
+    },
+  };
+
+  return (
+    <Content className={classes.content} data-testid="techdocs-content">
+      <Toolbar className={classes.contentToolbar}>
+        <Tooltip title={themes[theme].title} arrow>
+          <IconButton onClick={toggleTheme}>{themes[theme].icon}</IconButton>
+        </Tooltip>
+      </Toolbar>
```
[the remark this answers is not in the document]
Comment (camilaibs) [reply packages/techdocs-cli-embedded-app/src/components/TechDocsPage/TechDocsPage.tsx:120]: The current [TechDocsPageHeader](TechDocsPageHeader) does not accept labels...
Would it be a good idea to change it to accept labels in this pull request?

File: .changeset/techdocs-moose-design.md (hunk 1 above)
Comment (camilaibs) [reply .changeset/techdocs-moose-design.md:5]: Good idea, will do!

File: packages/techdocs-cli-embedded-app/src/components/TechDocsPage/TechDocsPage.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (camilaibs) [reply packages/techdocs-cli-embedded-app/src/components/TechDocsPage/TechDocsPage.tsx]: Fixed in this commit: https://github.com/backstage/backstage/pull/9569/commits/e2070bc7104101b1ee830264dba803898a4ec4c8

**Comment by camilaibs** [comment]
> > Thanks for the excellent instructions on how to test this PR!
> > I'm surprised to see things working with 3 major versions bumped 😄 however I noticed the resizing to smaller screens broke in a way that the sidebar overlaps with the content, see screenshot below:
> > Screenshot 2022-02-17 at 09 56 36
> 
> Thanks for testing @ottosichert , could you share the breakpoint you tested?

@ottosichert fixed here: https://github.com/backstage/backstage/pull/9569/commits/e2070bc7104101b1ee830264dba803898a4ec4c8.

File: .changeset/techdocs-moose-design.md (hunk 1 above)
Comment (camilaibs) [reply .changeset/techdocs-moose-design.md:5]: Updated, looks good to you?

**Review by ottosichert — APPROVED** [review]

File: .changeset/techdocs-moose-design.md (hunk 1 above)
Comment (emmaindal) [reply .changeset/techdocs-moose-design.md:5]: Looks great! I wonder if its enough to focus on the docker image and say something like:

> If you use the `spofify/techdocs` image to build your documentation, make sure you use version `spotify/techdocs:v0.3.7`.....

as not everyone uses the cli to generate + publish docs! (and the cli will use the new version per default)

**Pull request merged**